jhu-idc/idc-isle-dc
# Fix Collection Image Derivatives & Collection featured image issue.

#### General 
The things this PR addresses: 
  - Image derivatives were not being built automatically when put on Collection Objects. The thumbnail images are used in the feature collections section on the collection list page and the thumbnails are used in result lists.  This PR adds back in that feature. Resolves https://github.com/jhu-idc/idc-isle-dc/issues/267
  - Image derivatives were not being built automatically when put on a Paged Content object -- we should at least have thumbnails there. 
  - This pulls in the latest idc_ui_module changes:
       -  fixes an issue on the main collections list page where if a featured collection's thumbnail didn't have an image, it used the same one as the thumbnail before it.  (we've seen this before on a collection's landing page with featured items). 
       -  mutes a PHP notice that's innocuous
 - This pulls in the latest `idc_export` module which has some README changes as well as cleans up a few innocuous php notices.  


**Note: For whatever reason you cannot attach anything but the Image Media Type to a Collection Object. I'm not sure why it is this way - I didn't dig into this.  I figure that Image will be enough for the time being.**

#### To test:
For the Collection object derivative:
 - start up the system
  - create a Collection object
  - add an Image media type to the Collection object. 
  - verify that derivatives are created (right now it makes the usual derivatives (thumbnail, service file, fits) but we may not want the service file, since it's not used). 

For the Paged Content derivatives:
 - create a Paged Content item
 - add an Image media type to the Paged Content Object
 - verify that the derivs were created correctly. 
  
For the feature Collections section:
   - create a few more Collections with Image Media attached to them. 
   - mark a few as "featured"
   - create a Collection or two w/o an Image and mark them as "featured"
   - testing will be best if you have 4-5 collections and some do not have images. 
   - view the `/Collections` page and see the featured collections section. Ensure that the featured collection w/o an image does not have anything set for an image.  If your imageless featured collection is not after one that has an image, try to edit the collections until it shows up after one with an image. Ensure that the image is still unset.
